File: app/src/main/java/com/punchlines/punchline/paas/PaasService.java
package com.punchlines.punchline.paas;

import java.util.List;

import java8.util.concurrent.CompletableFuture;

public class PaasService {

    private PaasRepository repository;

    public PaasService(PaasRepository repository) {
        this.repository = repository;
    }

    public CompletableFuture<Punchline> dailyPunchline() {
        return repository.dailyPunchline();
    }

    public CompletableFuture<Punchline> randomPunchline() {
        return repository.randomPunchline();
    }

    public CompletableFuture<List<String>> artists() {
        return repository.artists();
    }

    public CompletableFuture<List<Punchline>> artistPunchlines(String artist) {
        return repository.artistPunchlines(artist);
    }

}

File: app/src/main/java/com/punchlines/punchline/paas/PaasService.kt
package com.punchlines.punchline.paas

import java8.util.concurrent.CompletableFuture

class PaasService(private val repository: PaasRepository) {

    fun dailyPunchline(): CompletableFuture<Punchline> =
        repository.dailyPunchline()

    fun randomPunchline(): CompletableFuture<Punchline> =
        repository.randomPunchline()

    fun artists(): CompletableFuture<List<String>> =
        repository.artists()

    fun artistPunchlines(artist: String): CompletableFuture<List<Punchline>> =
        repository.artistPunchlines(artist)

}